compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -360,7 +360,16 @@ class Namer { typer: Typer =>
   /** Expand tree and store in `expandedTree` */
   def expand(tree: Tree)(implicit ctx: Context): Unit = tree match {
     case mdef: DefTree =>
-      val expanded = desugar.defTree(mdef)
+      val expanded = mdef match {
+        case mdef: MemberDef if ctx.macrosEnabled && macros.hasAnnotMacros(mdef) =>
+          macros.expandAnnotMacro(mdef) match {
+            case Block(stats, _) =>                 // introducing of same level symbols
```

Comment:
The return code might be a block like `$class1; $object1` ([link](https://github.com/liufengyun/eden/blob/master/macros/src/main/scala/Annotations.scala#L29)). `Thicket` is not available in syntax.

Indeed dropping `expr` in blocks looks dubious. Here it involves a convention in annotation macros on how to introduce same level definitions. Only based on that protocol, we can drop `expr`. For safety reasons, some checks and warnings are in order here.